packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/ReviewWorkflows.js

```diff
@@ -1,35 +1,32 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useIntl } from 'react-intl';
+import { useSelector, useDispatch } from 'react-redux';
 import { SettingsPageTitle } from '@strapi/helper-plugin';
-import { Button, ContentLayout, HeaderLayout, Layout, Main } from '@strapi/design-system';
+import { Button, ContentLayout, HeaderLayout, Layout, Loader, Main } from '@strapi/design-system';
 import { Check } from '@strapi/icons';
 
 import { Stages } from './components/Stages';
+import { reducer } from './reducer';
+import { REDUX_NAMESPACE } from './constants';
+import { useInjectReducer } from '../../../../../../admin/src/hooks/useInjectReducer';
+import { useReviewWorkflows } from './hooks/useReviewWorkflows';
+import { setWorkflowLoadingState, setWorkflowStages } from './actions';
 
-const STAGES = [
-  {
-    uid: 'id-1',
-    name: 'To do',
-  },
-
-  {
-    uid: 'id-2',
-    name: 'Ready to review',
-  },
+export function ReviewWorkflowsPage() {
+  const { formatMessage } = useIntl();
+  const { workflows: workflowsData } = useReviewWorkflows();
+  const workflow = useSelector((state) => state?.[REDUX_NAMESPACE]);
+  const dispatch = useDispatch();
 
-  {
-    uid: 'id-3',
-    name: 'In progress',
-  },
+  useInjectReducer(REDUX_NAMESPACE, reducer);
 
-  {
-    uid: 'id-4',
-    name: 'Reviewed',
-  },
-];
+  useEffect(() => {
+    dispatch(setWorkflowLoadingState(workflowsData.status));
 
-export function ReviewWorkflowsPage() {
-  const { formatMessage } = useIntl();
+    if (workflowsData.status === 'success') {
+      dispatch(setWorkflowStages(workflowsData.data));
+    }
+  }, [workflowsData.status, workflowsData.data, dispatch]);
```

Comment:
Does the data _need_ to be put into redux for this? if it does then maybe the hook should be putting it into redux? 🤔  or can we just use the data from said hook?

Reply:
I didn't put it into the hook because e.g. in the CM it won't be coupled to redux and just takes care of the data loading. We can also use the plain data every time: in a loading state it would just be `null`. I don't have a strong preference for either direction ...

Reply:
Actually it makes the whole setup much simpler: https://github.com/strapi/strapi/pull/15635/commits/60ee442e357fb019c649b586526392c93facf7b2 👍🏼